sycl/test/lit.cfg.py

```diff
@@ -168,7 +168,7 @@
 
 if "amdgcn-amd-amdhsa" in triple:
     llvm_config.with_system_environment("ROCM_PATH")
-    config.available_features.add("hip_amd")
+    config.available_features.add("hip")
```

Comment:
Can we move this change to a separate commit or even PR?

Reply:
I don't see how, but maybe I don't fully understand the question.

Reply:
If I get it right, this change renames `hip_amd` lit feature into `hip`. The renaming makes sense to me, but it's not necessary for removing "HIP NVIDIA infrastructure support". The change contributes a significant part of changes to the diff, which will be easier to review separately.

Reply:
In practice that might be true, but in theory/semantically it's only possible after `hip_nvidia` removal. Also, the renaming part is only true for `sycl/test` while many of the changes are for `sycl/test-e2e`. There, it's not the renaming as `hip` LIT feature is pre-existing and means `hip` backend, which is true for both `hip_nvidia` and `hip_amd`. I can only change `hip_amd`->`hip` in tests' directives after/with the removal.

Reply:
I've updated this PR to only modify our tests. I think that is, essentially, what you've asked for.